charts/Makefile

```diff
@@ -2,35 +2,48 @@ OCP_VERSION ?= 4.19
 BRANCH ?= master
 SYNC2CHARTS ?= true
 ORGREPO ?= https://github.com/openshift
+PROJECTS = \
+	cluster-api \
+	cluster-api-provider-aws \
+	cluster-api-provider-openshift-assisted
 
-build-cluster-api-chart: PROJECT = cluster-api
-build-cluster-api-aws-chart: PROJECT = cluster-api-provider-aws
-build-cluster-api-azure-chart: PROJECT = cluster-api-provider-azure
+# Define specific overrides
+ORGREPO_cluster-api-provider-openshift-assisted = https://github.com/openshift-assisted
+KUSTOMIZE_CONFIG_DIRS_cluster-api-provider-openshift-assisted = "bootstrap" "controlplane"
 
 WKDIR ?= ../out
-BUILTDIR=$(WKDIR)/$(PROJECT)/config/tmp
 
-.PHONY: build
-build: build-cluster-api-chart build-cluster-api-aws-chart build-cluster-api-azure-chart
-
-.PHONY: build-cluster-api-chart
-build-cluster-api-chart:
-	WKDIR=$(WKDIR) ORGREPO=$(ORGREPO) PROJECT=$(PROJECT) BRANCH=$(BRANCH) ../scripts/build.sh
-	OCP_VERSION=$(OCP_VERSION) SYNC2CHARTS=$(SYNC2CHARTS) BUILTDIR="$(BUILTDIR)" PROJECT=$(PROJECT) ../scripts/sync2chart.sh
+.PHONY: build-cluster-api-chart build-cluster-api-provider-aws-chart build-cluster-api-provider-azure-chart build-cluster-api-provider-openshift-assisted-chart
+
+# Define the build target for each project
+define make-chart-target
+build-$(1)-chart:
+	@echo "Building $(1) chart"
+	@WKDIR="$(WKDIR)" \
+	ORGREPO="$(if $(ORGREPO_$(1)),$(ORGREPO_$(1)),$(ORGREPO))" \
+	KUSTOMIZE_CONFIG_DIRS="$(if $(KUSTOMIZE_CONFIG_DIRS_$(1)),$(KUSTOMIZE_CONFIG_DIRS_$(1)),)" \
+	PROJECT="$(1)" \
+	BRANCH="$(BRANCH)" \
+	../scripts/build.sh
+	@BUILTDIR="$(WKDIR)/$(1)/config/tmp" \
+	OCP_VERSION="$(OCP_VERSION)" \
+	SYNC2CHARTS="$(SYNC2CHARTS)" \
+	PROJECT="$(1)" \
+	../scripts/sync2chart.sh
+endef
+
+$(foreach proj,$(PROJECTS),$(eval $(call make-chart-target,$(proj))))
 
-.PHONY: build-cluster-api-aws-chart
-build-cluster-api-aws-chart:
-	WKDIR=$(WKDIR) ORGREPO=$(ORGREPO) PROJECT=$(PROJECT) BRANCH=$(BRANCH) ../scripts/build.sh
-	OCP_VERSION=$(OCP_VERSION) SYNC2CHARTS=$(SYNC2CHARTS) BUILTDIR="$(BUILTDIR)" PROJECT=$(PROJECT) ../scripts/sync2chart.sh
```

Comment:
This would've been repeated a third time, so it has been refactored to declare it dynamically. Unfortunately we cannot use pattern rule as we have components with multiple dashes, so we need to make it a bit more convoluted.